Add copy user ID button to profile, closes #281

Phillezi opened the pull request:
Hi!

I added a button to copy user ID in the profile page, like described in #281
It adds a snackbar entry message if it succeeds or fails. 

For example if user is null it will say:
![image](https://github.com/kthcloud/console/assets/112874974/129e13e5-24ef-49fd-b1de-37acbe949fc3)

or if 
```javascript
navigator.clipboard.writeText(user.id);
```
throws an error, it will show this message (error message is the message of the error thrown):
![Screenshot from 2024-05-22 01-11-51](https://github.com/kthcloud/console/assets/112874974/9014914a-721c-4f66-b4b3-50ddff3948af)

## Here's how it looks (in glorious GIF quality):

### English
![Screencast-from-2024-05-22-00-42-24](https://github.com/kthcloud/console/assets/112874974/ef42e504-f21a-4f91-b7b7-47441c627ca7)
### Swedish
![Screencast-from-2024-05-22-00-42-40](https://github.com/kthcloud/console/assets/112874974/a2572702-e327-44bc-b1a2-b9c6b0838698)

Lmk if I should change anything :)

pierrelefevre commented:
Nice! We do have a copy button component but since this is a bit isolated it should be fine :)

Phillezi commented:
> Nice! We do have a copy button component but since this is a bit isolated it should be fine :)

Oh, I saw it now, should i change to use that component instead?

pierrelefevre commented:
If you want to, sure! Would be a bit cleaner code 🚀

Phillezi commented:
Hi again!

I changed it now so that it uses the existing CopyButton component, I modified the CopyButton component so that it can be used with text intput (next to the icon) with the optional parameter textContent (it will only show if the variant is specified to not be icon too).

I also added that check-mark that I had before to this component, so it changes for 2s after you click it to show a check mark instead of the copy icon. 

Heres how it looks:

![Screencast-from-2024-05-22-12-40-27](https://github.com/kthcloud/console/assets/112874974/df695cb9-bcf1-4f7b-ab90-fcfa92298415)

On the existing copy ssh key
![Screencast-from-2024-05-22-12-40-44](https://github.com/kthcloud/console/assets/112874974/51d4d2f2-3f89-4370-a05c-1960e260b594)
